prow/scripts/build-generic.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+set -e
+
+readonly SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# shellcheck disable=SC1090
+source "${SCRIPT_DIR}/library.sh"
+
+usage () {
+    echo "Usage: \$ ${BASH_SOURCE[1]} /path/to/component"
+    exit 1
+}
+
+readonly SOURCES_DIR=$1
+
+if [[ -z "${SOURCES_DIR}" ]]; then
+    usage
+fi
+
+init
+
+if [ -n "${PULL_NUMBER}" ]; then
+  echo "Building from PR"
+  DOCKER_TAG="PR-${PULL_NUMBER}"
+elif [[ "${PULL_BASE_REF}" =~ release-.* ]]; then
+  echo "Building from release ${PULL_BASE_REF}"
+  DOCKER_TAG=$(cat "${SCRIPT_DIR}/../../RELEASE_VERSION")
+  echo "Reading docker tag from RELEASE_VERSION file, got: ${DOCKER_TAG}"
+
+  if [[ "${REPO_OWNER}" == "kyma-project" && "${REPO_NAME}" == "kyma" ]]; then
+    NEXT_RELEASE=$(cat "${SCRIPT_DIR}/../RELEASE_VERSION")
+    echo "Checking if ${NEXT_RELEASE} was already published on github..."
+    RESPONSE=$(curl -s -o /dev/null -w "%{http_code}" https://api.github.com/repos/kyma-project/kyma/releases/tags/"${NEXT_RELEASE}")
+    if [[ $RESPONSE != 404* ]]; then
```

Comment:
If GitHub returns a 500 we don't know if the release is there or not, if the API recovers in the meantime you'd release.

Reply:
If I understand it correctly we proceed only if the response is 404. In all the other cases we stope the execution.

Reply:
True, I don't know why I thought about this 500 path yesterday.